lib/phaxio/client.rb

```diff
@@ -271,15 +275,101 @@ def get_account_status
     def send_post(path, options)
       post(path, query: options.merge!({api_key: api_key, api_secret: api_secret}))
     end
+
+    # Public: Fail if the callback signature is not valid.
+    #
+    # Takes the same parameters as Phaxio.validate_callback_signature
+    #
+    # Raises Phaxio::InvalidCallbackSignature if valid_callback_signature
+    # returns false for the given options.
+    def validate_callback_signature(**options)
+      raise InvalidCallbackSignature unless valid_callback_signature?(**options)
+      true
+    end
+
+    # Public: Check the signature of the signed request.
+    #
+    # options - Type: hash. Information needed to authenticate the callback.
+    #   :url       - Type: string. The full URL that was called by Phaxio,
+    #                including the query. (required)
+    #   :params    - Type: hash. The POSTed form data (required)
+    #   :files     - Type: array. Submitted files (required - "received" fax
+    #                callback only)
+    #   :signature - Type: string. The X-Phaxio-Signature HTTP header value
+    #
+    # Returns true if the signature matches the signed request, otherwise false
+    def valid_callback_signature?(**options)
```

Comment:
Just for the sake of completion. You're right though, valid is all that's necessary.